ql/ql/src/queries/performance/LargeJoinOrder.ql

```diff
@@ -16,14 +16,36 @@ import KindPredicatesLog
 float getBadness(ComputeSimple simple) {
   exists(float maxTupleCount, float resultSize, float largestDependency, float denom |
     resultSize = simple.getResultSize() and
-    maxTupleCount = max(simple.getPipelineRun().getCount(_)) and
+    extractInformation(simple, maxTupleCount, _, _, _, _) and
     largestDependency = max(simple.getDependencies().getADependency().getResultSize()) and
     denom = resultSize.maximum(largestDependency) and
     denom > 0 and // avoid division by zero (which would create a NaN result).
     result = maxTupleCount / denom
   )
 }
 
-from ComputeSimple evt, float f
-where f = getBadness(evt) and f > 1.5
-select evt, f order by f desc
+pragma[nomagic]
+predicate hasPipelineRun(ComputeSimple simple, PipeLineRun run) { run = simple.getPipelineRun() }
+
+predicate extractInformation(
+  ComputeSimple simple, float maxTupleCount, int pipelineIndex, Array tuples,
+  Array duplicationPercentages, Array ra
+) {
+  exists(PipeLineRun run |
+    hasPipelineRun(simple, run) and
+    maxTupleCount = max(run.getCount(_)) and
+    pragma[only_bind_out](tuples.getFloat(pipelineIndex)) = maxTupleCount and
+    tuples = run.getCounts() and
+    duplicationPercentages = run.getDuplicationPercentage() and
+    ra = simple.getRa()
```

Comment:
Oh! You're right! Indeed, the `toString` on `Array` makes this ... slightly less useful.

Reply:
Hm, so which kind of format should we actually output here? Should we include a query predicate that outputs every RA operation in the pipeline?

Reply:
Something like
```
predicate_name, badness, ra_index, tuples[ra_index], duplication[ra_index], ra_text[ra_index]
```
seems to me like it would work pretty well. We can sort the lines by `predicate_name` and `ra_index` and then filter by `predicate_name` to reconstruct the information we need (and then elide the first three columns in the RA dump we thus produce).

Reply:
That duplicates the badness value for each RA operation in the pipeline, though

Reply:
True. I think we can live with that small bit of redundancy, though.

Alternatively we could output several separate tables (as query predicates), but this would complicate the code that needs to interpret this output.

Reply:
Alright, since this is a Hackathon I guess I can live with that redundancy 😂. I've implemented that in 96adbe13b1e8eaed5c2f0bdc412dc804fa71fb9b.

Reply:
That's the spirit! 😄